packages/@aws-cdk/aws-stepfunctions-tasks/lib/private/task-utils.ts

```diff
@@ -13,6 +14,22 @@ export function validatePatternSupported(integrationPattern: IntegrationPattern,
   }
 }
 
+/**
+ * Verifies that a received PropagatedTagSource is supported for a service integration
+ */
+export function validatePropagatedTagSourceSupported(
+  supportedPropagatedTagSources: PropagatedTagSource[],
+  propagatedTagSource?: PropagatedTagSource,
+) {
+  if (propagatedTagSource === undefined) {
+    return;
+  }
+
+  if (!supportedPropagatedTagSources.includes(propagatedTagSource)) {
+    throw new Error(`Unsupported service PropagatedTagSource. Supported PropagatedTagSources: ${supportedPropagatedTagSources}. Received: ${propagatedTagSource}`);
+  }
+}
+
```

Comment:
This validation isn't necessary. By making the the prop type an enum, this is already enforced.

Reply:
I added this validation by mistaken because I misunderstood you meant that there was no test case for the invalid value pattern in [the comment](https://github.com/aws/aws-cdk/pull/23537#discussion_r1072988100).
I'll remove this validation. Thanks!